### PR TITLE
fix: Fix retries by ensuring string error names get converted to their integer codes

### LINKED ISSUE #502: Retries are not working because incoming configs use string gRPC codes while gRPC exceptions use numeric codes
AFAICT _no_ retries are currently working in the generated clients. This is because input grpc_service_config.json files use the string forms of gRPC error codes (e.g. `"UNAVAILABLE"` or `"DEADLINE_EXCEEDED"`) and the generator is copying those as strings into the default RPC configs. However, the retry logic in gapic-common is comparing those string codes with the _numeric_ codes from `GRPC::BadStatus#code`. As a result, the retry logic thinks that no error is retryable.

We should fix this in two places. (Technically, only one or the other is sufficient, but for safety I think it's better to do both.)
* `Gapic::CallOptions::RetryPolicy` should include logic that maps string retry codes to their numeric equivalents before comparing them with the incoming gRPC error code.
* When the generator generates initial RPC configs from the grpc_service_config.json data, it should map string codes to numeric codes.

The mapping is defined at https://developers.google.com/maps-booking/reference/grpc-api/status_codes

We will release a fixed gapic-common gem, and a fixed generator. This will regenerate all versioned gems.

### PR DESCRIPTION
**Pull request by dazuma**
Fixes #502

### CODE REVIEW

File: gapic-common/test/gapic/call_options/retry_policy/settings_test.rb [hunk 1]
```diff
@@ -74,4 +74,17 @@ def test_apply_defaults_wont_override_custom_values
     assert_equal 6, retry_policy.multiplier
     assert_equal 5, retry_policy.max_delay
   end
+
```
Comment (quartzmo) [inline gapic-common/test/gapic/call_options/retry_policy/settings_test.rb:77]: What do you think about expanding this to cover all 16 error codes, with a separate test for each code?

File: gapic-common/lib/gapic/call_options/retry_policy.rb [hunk 2]
```diff
@@ -77,14 +77,39 @@ def call error
...
+        "ALREADY_EXISTS",
+        "PERMISSION_DENIED",
+        "RESOURCE_EXHAUSTED",
+        "FAILED_PRECONDITION",
+        "ABORTED",
+        "OUT_OF_RANGE",
+        "UNIMPLEMENTED",
+        "INTERNAL",
+        "UNAVAILABLE",
+        "DATA_LOSS"
```
Comment (quartzmo) [inline gapic-common/lib/gapic/call_options/retry_policy.rb:105]: What about `UNAUTHENTICATED` - `16` in this [list](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)?

Comment (dazuma) [reply gapic-common/lib/gapic/call_options/retry_policy.rb:105]: Ah, I didn't see that list. I wouldn't expect `UNAUTHENTICATED` ever to be retryable, but I'll add it for completeness.

File: gapic-common/test/gapic/call_options/retry_policy/settings_test.rb (hunk 1 above)
Comment (dazuma) [reply gapic-common/test/gapic/call_options/retry_policy/settings_test.rb:77]: Sure, I can do that.

File: gapic-common/lib/gapic/call_options/retry_policy.rb (hunk 2 above)
Comment (quartzmo) [reply gapic-common/lib/gapic/call_options/retry_policy.rb:105]: Want to add a link that list in comments here?

File: gapic-common/test/gapic/call_options/retry_policy/settings_test.rb (hunk 1 above)
Comment (dazuma) [reply gapic-common/test/gapic/call_options/retry_policy/settings_test.rb:77]: Done. Also fixed the name of one of the test methods (I forgot methods have to be named `test_*`.)

File: gapic-common/lib/gapic/call_options/retry_policy.rb (hunk 2 above)
Comment (dazuma) [reply gapic-common/lib/gapic/call_options/retry_policy.rb]: Done.

Comment (dazuma) [reply gapic-common/lib/gapic/call_options/retry_policy.rb]: And added link.

**Pull request merged**